### PR TITLE
Added SPV createHtlc RPC

### PR DESCRIPTION
**Pull request by Jouzo**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:

Adds SPV createHtlc RPC

#### Which issue(s) does this PR fixes?:

Fixes part of #202

### CODE REVIEW

File: packages/jellyfish-api-core/src/category/spv.ts [hunk 1]
```diff
@@ -55,6 +57,20 @@ export class Spv {
...
     return await this.client.call('spv_sendtoaddress', [address, amount, options.feeRate], 'bignumber')
   }
+
+  /**
+   * Creates a Bitcoin address whose funds can be unlocked with a seed or as a refund.
+   *
+   * @param {string} receiverPubKey The public key of the possessor of the seed
+   * @param {number} ownerPubKey The public key of the recipient of the refund
+   * @param {CreateHtlcOptions} options
+   * @param {string} options.timeout  Timeout of the contract (denominated in blocks) relative to its placement in the blockchain. Minimum 9. See HTLC_MINIMUM_BLOCK_COUNT
```
Comment (canonbrother) [inline packages/jellyfish-api-core/src/category/spv.ts:67]:
```suggestion
   * @param {string} [options.timeout='9']  Timeout of the contract (denominated in blocks) relative to its placement in the blockchain. Minimum 9. See HTLC_MINIMUM_BLOCK_COUNT
```

File: packages/jellyfish-api-core/src/category/spv.ts [hunk 2]
```diff
@@ -55,6 +57,20 @@ export class Spv {
   async sendToAddress (address: string, amount: BigNumber, options: SendToAddressOptions = { feeRate: new BigNumber('10000') }): Promise<SendMessageResult> {
     return await this.client.call('spv_sendtoaddress', [address, amount, options.feeRate], 'bignumber')
   }
+
+  /**
+   * Creates a Bitcoin address whose funds can be unlocked with a seed or as a refund.
+   *
+   * @param {string} receiverPubKey The public key of the possessor of the seed
+   * @param {number} ownerPubKey The public key of the recipient of the refund
```
Comment (canonbrother) [inline packages/jellyfish-api-core/src/category/spv.ts:65]:
```suggestion
   * @param {string} ownerPubKey The public key of the recipient of the refund
```

File: packages/jellyfish-api-core/__tests__/category/spv/createHtlc.test.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    const promise = client.spv.createHtlc(pubKeyA, pubKeyB, { timeout: `${HTLC_MINIMUM_BLOCK_COUNT - 1}` })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow("RpcApiError: 'Timeout below minimum of 9', code: -3, method: spv_createhtlc")
+  })
+
+  it('should not createHtlc with denominated relative timeout', async () => {
+    const pubKeyA = await container.call('spv_getaddresspubkey', [await container.call('spv_getnewaddress')])
+    const pubKeyB = await container.call('spv_getaddresspubkey', [await container.call('spv_getnewaddress')])
+
+    const promise = client.spv.createHtlc(pubKeyA, pubKeyB, { timeout: '4194304' })
```
Comment (canonbrother) [inline packages/jellyfish-api-core/__tests__/category/spv/createHtlc.test.ts:69]: help add a minor note here, easy to review and understand
```suggestion
    const promise = client.spv.createHtlc(pubKeyA, pubKeyB, { timeout: '4194304' }) // SEQUENCE_LOCKTIME_TYPE_FLAG = (1 << 22)
```

**Review by canonbrother — COMMENTED** [review]
overall good for me

File: packages/jellyfish-api-core/src/category/spv.ts [hunk 4]
```diff
@@ -1,6 +1,8 @@
 import { ApiClient } from '../.'
 import BigNumber from 'bignumber.js'
 
+export const HTLC_MINIMUM_BLOCK_COUNT = 9
+
 /**
  * SPV RPCs for DeFi Blockchain
```
Comment (surangap) [inline packages/jellyfish-api-core/src/category/spv.ts:7]: these SPV RPCs are for Bitcoin network(for now). maybe rename to `SPV RPCs for Bitcoin Blockchain`
open for discussion. @fuxingloh

File: packages/jellyfish-api-core/__tests__/category/spv/createHtlc.test.ts [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should createHtlc', async () => {
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/spv/createHtlc.test.ts:19]: create a similar test `should createHtlc with given seed`

File: packages/jellyfish-api-core/src/category/spv.ts [hunk 6]
```diff
@@ -78,3 +94,21 @@ export interface SendMessageResult {
...
+  seed?: string
+}
+
+export interface CreateHtlcResult {
+  /** The value of the new Bitcoin address */
+  address: string
+  /** Hex-encoded redemption script */
+  redeemScript: string
+  /** Hex-encoded seed */
+  seed: string
```
Comment (surangap) [inline packages/jellyfish-api-core/src/category/spv.ts:111]: do this https://github.com/DeFiCh/jellyfish/pull/511/files#r680662678 and check whether `seed` is optional. and update accordingly.

File: packages/jellyfish-api-core/src/category/spv.ts (hunk 4 above)
Comment (fuxingloh) [reply packages/jellyfish-api-core/src/category/spv.ts:7]: Yes, this make sense

**Review by surangap — CHANGES_REQUESTED** [review]
Request to add some more tests and docs changes

File: packages/jellyfish-api-core/src/category/spv.ts (hunk 6 above)
Comment (Jouzo) [reply packages/jellyfish-api-core/src/category/spv.ts:111]: good catch ! Updated the interface and set seed and seedhash as optionals

**Review by surangap — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**